torch/nn/modules/lazy.py

```diff
@@ -174,7 +174,7 @@ def __init__(self: _LazyProtocol, *args, **kwargs):
         # Mypy doesnt like this super call in a mixin
         super().__init__(*args, **kwargs)  # type: ignore[misc]
         self._load_hook = self._register_load_state_dict_pre_hook(self._lazy_load_hook)
-        self._initialize_hook = self.register_forward_pre_hook(self._infer_parameters)
+        self._initialize_hook = self.register_forward_pre_hook(self._infer_parameters, with_kwargs=True)
```

Comment:
This would be BC-breaking no?
Even for simple things like the LazyLinear, this will make it fail if any argument is passed as a kwargs.